### PR TITLE
feat(skymp5-server): add rate and mult actor values support & add basic magic effects support

### LINKED ISSUE #1498: Add rate and mult actor values support

### PR DESCRIPTION
**Pull request by che6a**
closes #1498

### CODE REVIEW

**Review by Pospelove — CHANGES_REQUESTED** [review]
this timer will not save across server restarts

I think we should store active magic effects in changeform somehow

File: viet/src/Timer.cpp [hunk 1]
```diff
@@ -54,3 +43,42 @@ void Viet::Timer::TickTimers()
     front.promise.Resolve(Viet::Void());
   }
 }
+
+bool Viet::Timer::RemoveTimer(
+  const std::chrono::system_clock::time_point& endTime)
+{
+  auto& timers = pImpl->timers;
+  std::sort(timers.begin(), timers.end(),
```
Comment (Pospelove) [inline viet/src/Timer.cpp:51]: remove sort

**Review by Pospelove — CHANGES_REQUESTED** [review]

File: skymp5-server/cpp/server_guest_lib/ActiveMagicEffectsMap.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+{
+  auto res = nlohmann::json::array();
+  for (const auto& [actorValue, effectEntry] : effects) {
+    auto obj = nlohmann::json::object();
+    obj["effectId"] = effectEntry.data.effectId;
+    obj["endTime"] = TimeUtils::ToString(effectEntry.endTime);
+    obj["magnitude"] = effectEntry.data.magnitude;
+    obj["duration"] = effectEntry.data.duration;
+    obj["areaOfEffect"] = effectEntry.data.areaOfEffect;
+    obj["actorValue"] = static_cast<int32_t>(actorValue);
```
[the remark this answers is not in the document]
Comment (che6a) [reply skymp5-server/cpp/server_guest_lib/ActiveMagicEffectsMap.cpp:65]: std::to_underlying is available only since C++23

File: unit/PapyrusFormTest.cpp [hunk 3]
```diff
@@ -10,7 +11,11 @@ TEST_CASE("RegisterForSingleUpdate", "[Papyrus][Form]")
   class CustomForm : public MpForm
   {
   public:
-    void Update() override { counter++; }
+    void Update() override
+    {
+      counter++;
+      std::cout << "Updated!\n";
```
Comment (Pospelove) [inline unit/PapyrusFormTest.cpp:17]: why is this needed?

**Review by Pospelove — CHANGES_REQUESTED** [review]

**Comment by Pospelove** [comment]
Overall seems good. needs testing by SupAidmi

File: unit/PapyrusFormTest.cpp (hunk 3 above)
Comment (che6a) [reply unit/PapyrusFormTest.cpp:17]: It's been needed for testing and shouldn't have been there :) I'll remove it today.

**Review by Pospelove — CHANGES_REQUESTED** [review]
```
[2023-07-05 19:39:12.187] [console] [error] Parsing of world/changeForms/5c59d_CombatSettings.esp.json failed with Unable to read key 'effects' from JSON element '{"angle":[0.0,0.0,326.58758544921875],"appearanceDump":null,"baseContainerAdded":false,"baseDesc":"bad9a:CombatSettings.esp","dynamicFields":{},"equipmentDump":null,"formDesc":"5c59d:CombatSettings.esp","healthPercentage":1.0,"inv":{"entries":[]},"isDead":false,"isDisabled":false,"isHarvested":true,"isOpen":false,"isRaceMenuOpen":false,"magickaPercentage":1.0,"nextRelootDatetime":1687540985,"position":[76398.8203125,-60395.296875,10560.220703125],"profileId":-1,"recType":0,"spawnDelay":5.0,"spawnPoint_cellOrWorldDesc":"3c:Skyrim.esm","spawnPoint_pos":[133857.0,-61130.0,14662.0],"spawnPoint_rot":[0.0,0.0,72.0],"staminaPercentage":1.0,"worldOrCellDesc":"3c:Skyrim.esm"}': NO_SUCH_FIELD: The JSON field referenced does not exist in this object.
```
characters from previous server versions (and possible all objects) stopped working. please make `effects` in changeform optional

**Review by Pospelove — APPROVED** [review]

**Comment by Pospelove** [comment]
needs testing
